drivers: usb: device: Migrate to new logger

ydamigos opened the pull request:
Move to new logger subsystem.

Signed-off-by: Yannis Damigos <giannis.damigos@gmail.com>

ydamigos committed:
drivers: usb: device: Migrate to new logger

Move to new logger subsystem.

Signed-off-by: Yannis Damigos <giannis.damigos@gmail.com>

ydamigos commented:
Updated to resolve conflicts.

galak reviewed (dismissed):
Can you fix this for the change in LOG_MODULE_REGISTER to take the name as a param, and remove LOG_MODULE_NAME

ydamigos commented:
@galak Done

ydamigos commented:
recheck

ydamigos commented:
Updated to migrate all usb device drivers to new logger.

ydamigos commented:
@finikorg @jfischer-phytec-iot @galak @loicpoulain @erwango Could you check it again?

finikorg commented on drivers/usb/device/usb_dc_dw.c:
```diff
@@ -587,7 +588,7 @@ static void usb_dw_handle_enum_done(void)
 
-	SYS_LOG_DBG("USB ENUM DONE event, %s speed detected",
+	LOG_DBG("USB ENUM DONE event, %s speed detected",
 	    speed == USB_DW_DSTS_ENUM_LS ? "Low" : "Full");
```
all multi-line logs need to be adjusted

ydamigos replied:
done

finikorg commented on drivers/usb/device/usb_dc_kinetis.c:
```diff
@@ -183,7 +184,7 @@ static int kinetis_usb_init(void)
 	irq_enable(CONFIG_USBD_KINETIS_IRQ);
 
-	SYS_LOG_DBG("");
+	LOG_DBG("");
```
I assume this does not print function name, right? I think we need to invent some macro like networking and other subsys are doing to overcome missing function name limitation.

ydamigos replied:
Thanks for the pointer. Done.

finikorg reviewed (changes requested):
Main issues are: multi-line adjustments and missing function name. Look what I have done here: 9d5c8402fec1faf3cb2a85ca8aef7efe0afed319

finikorg commented on drivers/usb/device/usb_dc_dw.c:
```diff
@@ -653,7 +645,7 @@ static void usb_dw_isr_handler(void)
 
-			SYS_LOG_DBG("USB OUT EP%d: RX_FLVL status %d, size %d",
+			LOG_DBG("USB OUT EP%d: RX_FLVL status %d, size %d",
 				ep_idx, status, xfer_size);
```
Does this line need adjustment?

finikorg replied:
It also makes sense to add function name to the logs.

finikorg replied:
Or, maybe, conversion LOG_DBG => USB_DBG might be done in the follow up patch, just fix adjustment.

ydamigos replied:
I added USB_* macros.

finikorg reviewed (commented):
What about defining
USB_DBG()
USB_WRN()
USB_ERR()
the same way networking subsys does. See faf3068f6da9b1fb606cb47c2cb8530134c6ead4.
Better to use the same size for macros for easy adjustment of multi-line logs (i.e. LOG_DBG => USB_DBG)

jukkar commented:
Just fyi and not really relevant to this PR, but I have USB NET conversion to new logging at https://github.com/zephyrproject-rtos/zephyr/pull/8816/commits/4a89de21712fb8f4524b55072ba11c0be8d898a5

finikorg reviewed (approved):
Thanks, looks good, USB_* macros might be put later to some common header.

jfischer-phytec-iot commented on samples/subsys/usb/cdc_acm/prj.conf:
```diff
@@ -4,7 +4,7 @@ CONFIG_GPIO=y
 CONFIG_USB_DEVICE_STACK=y
 CONFIG_USB_DEVICE_PRODUCT="Zephyr CDC ACM sample"
-CONFIG_SYS_LOG_USB_DRIVER_LEVEL=1
+CONFIG_USB_DRIVER_LOG_LEVEL=1
```
Please add CONFIG_LOG=y as well to all prj*.conf

ydamigos replied:
Done

ydamigos commented:
Shippable is not happy because `samples/net/zperf` tests fail on quark_se_c1000_devboard due to RAM overflow.
@finikorg I don't like to disable tests on quark_se_c1000_devboard for `samples/net/zperf`, is there another way to make shippable happy?

finikorg committed:
samples: zperf: Fix sanity check with NET, USB and logs

Decrease number of buffer for sanity check, in real sample logs
consuming a lot of memory might be traded for buffers. The sample was
failing for quark_se_c1000_devboard.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

finikorg commented:
@ydamigos Fixed.

BTW: @ydamigos @jfischer-phytec-iot @loicpoulain @carlescufi Is somebody is working already on converting USB Device stack to the new logger? I can do this but do not like to do double work.

loicpoulain commented:
> Is somebody is working already on converting USB Device stack to the new logger? 

@ydamigos, not on my side.

ydamigos commented:
> Is somebody is working already on converting USB Device stack to the new logger? I can do this but do not like to do double work.

@findlayfeng Thanks. I already created PR #10168.

ydamigos commented:
Updated to resolve conflicts.

nashif commented on drivers/usb/device/usb_dc_kinetis.c:
```diff
@@ -183,7 +189,7 @@ static int kinetis_usb_init(void)
 	irq_enable(CONFIG_USBD_KINETIS_IRQ);
 
-	SYS_LOG_DBG("");
+	USB_DBG("");
```
do we really need those? why?

nashif commented on drivers/usb/device/usb_dc_kinetis.c:
```diff
@@ -216,15 +222,15 @@ int usb_dc_reset(void)
 		       USB_INTEN_ERROREN_MASK);
 
-	SYS_LOG_DBG("");
+	USB_DBG("");
```
do we really need those? why?

nashif commented on drivers/usb/device/usb_dc_kinetis.c:
```diff
@@ -246,7 +252,7 @@ int usb_dc_attach(void)
 int usb_dc_detach(void)
 {
-	SYS_LOG_DBG("");
+	USB_DBG("");
```
ditto

nashif commented on drivers/usb/device/usb_dc_dw.c:
```diff
@@ -25,8 +25,15 @@
+#include <logging/log.h>
+LOG_MODULE_REGISTER(usb_dc_dw);
+
+#define USB_DBG(fmt, ...) LOG_DBG("(%p): %s: " fmt, k_current_get(), \
```
why are we redefining this here? is it to get the function/line no only? We should avoid this deviation and introduction of subsystem specific macros and instead look how we can get this functionality from the core logger. @nordic-krch, is there a way for us to include __func__ in the output in a generic way?

ydamigos commented:
@nashif All your comments are answered in PR #10168. I am closing this PR in favor of #10168.
